app/com/yetu/oauth2provider/oauth2/services/AuthorizeService.scala

```diff
@@ -35,7 +37,7 @@ class AuthorizeErrorHandler(clientService: IClientService,
     AuthorizeRequest(authorization.headers, authorization.params)
   }
 
-  def handleAuthorizeRequest[A](request: AuthorizeRequest, user: YetuUser): Either[OAuthError, AuthorizedClient] = try {
+  def handleAuthorizeRequest[A](request: AuthorizeRequest, user: YetuUser): Future[Either[OAuthError, AuthorizedClient]] = try {
```

Comment:
I'm not sure how the API works, but instead of a `Future[Either[OAuthError, AuthorizedClient]]` I would model it as a `Future[AuthorizedClient]` and use `Future.failed` for the error cases